### PR TITLE
[LETS-665] page server served compressed page type profiling

### PR DESCRIPTION
**Pull request by cristiarg**
http://jira.cubrid.org/browse/LETS-665

Add new complex perfstat infrastructure for types of pages that are dispatched by a Page Server to the connected Transaction Servers - `PSTAT_SERVED_COMPR_PAGE_TYPE_COUNTERS` .
The counters are entered across the following dimensions:
- page type: `PERF_PAGE_TYPE`
- newly introduced `PERF_SERVED_COMPR_PAGE_TYPE` which has 3 values:
  - count of compressed and transferred pages of that type
  - accumulated ratio of pages of that type (registered as multiplied by a factor of `100.0`, because the current perfstat infrastructure does not allow non-integeres); when these values are printed out, the accumulated ratio is averaged and divided by `100.0` (see implementation function `f_dump_in_buffer_served_compr_page_type_impl`)

This logging is guarded by a detailed logging flag - `PERFMON_ACTIVATION_FLAG_SERVED_COMPR_PAGE_TYPE`.
The infrastructure is used in function `pgbuf_respond_data_fetch_page_request`.

### CODE REVIEW

File: src/base/perf_monitor.h [hunk 1]
```diff
@@ -268,6 +276,14 @@ typedef enum
   PERF_SNAPSHOT_VISIBILITY_CNT
 } PERF_SNAPSHOT_VISIBILITY;
 
+typedef enum
+{
+  PERF_SERVED_COMPR_COUNT = 0,	/* per page type, this many pages have been compressed and served by page server */
+  PERF_SERVED_COMPR_RATIO,	/* per page type, the average ratio of pages compressed and served by page server */
+
+  PERF_SERVED_COMPR_CNT
```
Comment (razvanradulescu) [inline src/base/perf_monitor.h]: Nit : naming : PERF_SERVED_COMPR_CNT-> PERF_SERVED_COMPR_PAGE_TYPE_CNT (PERF_SERVED_COMPR_CNT can be confused with PERF_SERVED_COMPR_COUNT)

File: src/base/perf_monitor.h [hunk 2]
```diff
@@ -134,6 +135,13 @@ typedef enum
 #define PERF_OBJ_LOCK_STAT_COUNTERS (SCH_M_LOCK + 1)
 #define PERF_DWB_FLUSHED_BLOCK_VOLUMES_CNT 10
 
+/* PAGE_TYPE x (PAGE_COUNT + COMPR_RATIO_x_100) */
+#define PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_SIZE \
+  (PERF_PAGE_CNT * PERF_SERVED_COMPR_CNT)
+
+#define PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_OFFSET(page_type, count_or_compress_ratio_index) \
+  ((page_type) * 2) + (count_or_compress_ratio_index)
```
Comment (razvanradulescu) [inline src/base/perf_monitor.h]:
```suggestion
  ((page_type) * PERF_SERVED_COMPR_CNT) + (count_or_compress_ratio_index)
```

File: src/base/perf_monitor.c [hunk 3]
```diff
@@ -4325,3 +4332,136 @@ perfmon_er_log_current_stats (THREAD_ENTRY * thread_p)
...
+    char buf[BUF_LEN_MAX];	// declare once buffer and reuse each iteration below
+
+    assert (f_ptr != nullptr);
+    for (int page_type = PERF_PAGE_UNKNOWN; page_type < PERF_PAGE_CNT; ++page_type)
+      {
+	const int offset = PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_OFFSET (page_type, PERF_SERVED_COMPR_COUNT);
+	assert (offset < PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_SIZE);
+	assert ((offset + PERF_SERVED_COMPR_RATIO) < PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_SIZE);	// offset for accumulated ratio
+
+	const UINT64 counter = stats_ptr[offset];
```
Comment (razvanradulescu) [inline src/base/perf_monitor.c]:
```suggestion
	const UINT64 counter = stats_ptr[offset + PERF_SERVED_COMPR_COUNT];
```

File: src/base/perf_monitor.c [hunk 4]
```diff
@@ -4325,3 +4332,136 @@ perfmon_er_log_current_stats (THREAD_ENTRY * thread_p)
...
+
+  if (*s != nullptr)
+    {
+      for (int page_type = PERF_PAGE_UNKNOWN; page_type < PERF_PAGE_CNT; ++page_type)
+	{
+	  const int offset = PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_OFFSET (page_type, PERF_SERVED_COMPR_COUNT);
+	  assert (offset < PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_SIZE);
+	  assert ((offset + PERF_SERVED_COMPR_RATIO) < PERF_SERVED_COMPR_PAGE_TYPE_COUNTERS_SIZE);	// offset for accumulated ratio
+
+	  const UINT64 counter = stats_ptr[offset];
```
Comment (razvanradulescu) [inline src/base/perf_monitor.c]:
```suggestion
	  const UINT64 counter = stats_ptr[offset + PERF_SERVED_COMPR_COUNT];
```

**Review by razvanradulescu — APPROVED** [review]

**Review by hornetmj — APPROVED** [review]

**Review by wrlawodms — APPROVED** [review]

**Review by joohok — APPROVED** [review]

**Pull request merged**